rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java

```diff
@@ -569,4 +575,80 @@ public void fillCheckedAttachmentData(Attachment attachment, User user) throws S
             attachment.unsetCheckedOn();
         }
     }
+
+    public boolean isValidSbomFile(MultipartFile file) throws TException {
+
+        if (file == null || file.isEmpty()) {
+            return false;
+        }
+
+        String originalFilename = file.getOriginalFilename();
+        File tmpFile = null;
+
+        try {
+            String lowerCaseFilename = originalFilename.toLowerCase();
+
+            // SPDX formats
+            if (lowerCaseFilename.endsWith(".spdx") || lowerCaseFilename.endsWith(".rdf")) {
+
+                String extension = lowerCaseFilename.endsWith(".spdx") ? ".spdx" : ".rdf";
+                tmpFile = saveAsTempFile(file.getInputStream(), "sbom_validation", extension);
+
+                try {
+                    SpdxDocument doc = deserializeDocument(tmpFile);
+
+                    String specVersion = doc.getSpecVersion();
+                    if (specVersion == null || specVersion.isEmpty()) {
+                        log.debug("SPDX document missing SPDX specification version");
+                        return false;
+                    }
+
+                    String documentUri = doc.getDocumentUri();
+                    if (documentUri == null || documentUri.isEmpty()) {
+                        log.debug("SPDX document missing required Document URI/namespace");
+                        return false;
+                    }
+
+                    return true;
+
+                } catch (InvalidSPDXAnalysisException | InvalidFileNameException e) {
+                    log.debug("Not a valid SPDX file: {}", e.getMessage());
+                    return false;
+                }
+            }
+
+            // CycloneDX formats
+            else if (lowerCaseFilename.endsWith(".xml")) {
```

Comment:
SPDX and CycloneDX, both formats can be serialized to xml and json files. It is better to take the `type` as an input from the controller class to determine which parser/loader to use.